src/messages/direct.rs

```diff
@@ -71,17 +70,29 @@ pub enum BootstrapResponse {
     Error(BootstrapResponseError),
 }
 
+/// Request to join a section
+#[cfg_attr(feature = "mock_serialise", derive(Clone))]
```

Comment:
Why not derive `Clone` unconditionally?

Reply:
Followed BootstrapResponse pattern. I think the main thing is DirectMessage should not be clone in production.
But probably these could be cleaned up as they could be clone maybe unconditionaly, Probably do that another time though.